client/src/components/AppCommandBar/AppCommandBar.test.js

```diff
@@ -0,0 +1,171 @@
+import AppCommandBar from "./AppCommandBar";
+
+import { configService } from "../../services/ConfigService";
+import { apiService } from "../../services/ApiService";
+import { eventService } from "../../services/EventService";
+
+import initIcons from "../../services/IconService/IconService";
+
+initIcons();
+
+jest.mock("../../services/ConfigService");
+jest.mock("../../services/ApiService");
+jest.mock("../../services/EventService");
+
+const getAllTwins = jest.spyOn(apiService, "getAllTwins");
+const setConfig = jest.spyOn(configService, "setConfig");
+
+
+
+const mockSuccesResponse = { "Status": "Success"};
+
+const optionalComponentsState = [{
+    id: "console",
+    name: "Console",
+    show: false,
+    showProp: "showConsole"
+  },{
+    id: "output",
+    name: "Output",
+    show: false,
+    showProp: "showOutput"
+  }
+];
+
+let toggleCounter = 0;
+
+const toggleOptionalComponent = () => {
+  toggleCounter++;
+};
+
+let container = null;
+beforeEach(() => {
+  container = document.createElement("div");
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  PubSub.clearAllSubscriptions();
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+
+test("delete all twins component", async () => {
+    // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+    configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+    apiService.getAllTwins.mockResolvedValue(mockSuccesResponse);
+
+    act(() => {render(<AppCommandBar optionalComponents={optionalComponentsState}
+        optionalComponentsState={optionalComponentsState}/>, container);});
+    
+    const button = container.querySelector(".delete-button");
+    act(() => {
+      button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+    });
+
+    const deleteButton = await screen.findByTestId("deleteTwins");
+
+    act(() => {
+      deleteButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+    });
+
+    expect(getAllTwins).toHaveBeenCalledTimes(1);
+
+  });
+
+  test("switch environment", async () => {
+    // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+    configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+    configService.setConfig.mockResolvedValue(mockSuccesResponse);
+
+    act(() => {render(<AppCommandBar optionalComponents={optionalComponentsState}
+        optionalComponentsState={optionalComponentsState}/>, container);});
+    
+    const button = container.querySelector(".sign-in");
+    act(() => {
+      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+    });
+
+    let input = await screen.findByTestId("urlInput");
+    input.value = "https://test.api.scus.digitaltwins.azure.net";
+    Simulate.change(input);
+    expect(input.value).toBe("https://test.api.scus.digitaltwins.azure.net");
```

Comment:
We probably don't need this test. Setting an input value and then checking that it was set isn't probably going to break that much?